### PR TITLE
Auto create pivot permissions cleanup

### PR DESCRIPTION
**Pull request by mourya-33**
### Feature or Bugfix
- Bugfix

### Detail
This PR is to restrict few overly permissive policy permissions on the auto created Pivot Role

### Relates
[- <URL or Ticket>](https://github.com/data-dot-all/dataall/issues/875)

### Security
Please answer the questions below briefly where applicable, or write `N/A`. Based on
[OWASP 10](https://owasp.org/Top10/en/).

- Does this PR introduce or modify any input fields or queries - this includes
fetching data from storage outside the application (e.g. a database, an S3 bucket)? No
  - Is the input sanitized? N/A
  - What precautions are you taking before deserializing the data you consume? N/A
  - Is injection prevented by parametrizing queries?N/A
  - Have you ensured no `eval` or similar functions are used?N/A
- Does this PR introduce any functionality or component that requires authorization?N/A
  - How have you ensured it respects the existing AuthN/AuthZ mechanisms?N/A
  - Are you logging failed auth attempts?N/A
- Are you using or adding any cryptographic features?N/A
  - Do you use a standard proven implementations?Yes
  - Are the used keys controlled by the customer? Where are they stored?N/A
- Are you introducing any new policies/roles/users?N/A
  - Have you used the least-privilege principle? How?N/A

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by mourya-33** [comment]
Tested the following

1. Linking environments
2. Unlinking environments
3. Add datasets, delete datasets
4. ML Studio and Notebooks

File: backend/dataall/modules/datapipelines/cdk/pivot_role_datapipelines_policy.py [hunk 1]
```diff
@@ -50,7 +50,7 @@ def get_statements(self):
                     "cloudformation:DescribeStackResources"
                 ],
                 resources=[
-                    f'arn:aws:cloudformation:*:{self.account}:stack/*/*',
+                    f'arn:aws:cloudformation:*:{self.account}:stack/{self.env_resource_prefix}*/*',
```
Comment (noah-paige) [inline backend/dataall/modules/datapipelines/cdk/pivot_role_datapipelines_policy.py:53]: Isn't this permission already granted in `/dataall/core/environment/cdk/pivot_role_core_policies/cloudformation.py`?

Do we need this permission then for data.all pipelines?

**Comment by noah-paige** [comment]
Request - add changes to `pivotRole.yaml` as well

File: backend/dataall/modules/datapipelines/cdk/pivot_role_datapipelines_policy.py (hunk 1 above)
Comment (mourya-33) [reply backend/dataall/modules/datapipelines/cdk/pivot_role_datapipelines_policy.py]: @noah-paige i removed this permission in the data pipelines as this is already covered under core policy

**Comment by mourya-33** [comment]
> Request - add changes to `pivotRole.yaml` as well

Done @noah-paige 

Changes:
1. removed DeleteApp permission for sagemaker as the Apps are not created by pivotRole
2. removed cloudformation stack permissions for pipelines, as this is already covered under the core cloudformation policy for pivot role

File: deploy/pivot_role/pivotRole.yaml [hunk 2]
```diff
@@ -213,7 +213,8 @@ Resources:
             Action:
               - 'logs:PutLogEvents'
             Effect: Allow
-            Resource: '*'
+            Resource: 
+              - !Sub 'arn:aws:logs:*:${AWS::AccountId}:log-group:/${EnvironmentResourcePrefix}*'
```
Comment (noah-paige) [inline deploy/pivot_role/pivotRole.yaml:217]: In CDK Pivot Role we have the resource as `arn:aws:logs:*:{self.account}:log-group:*` but here it is different

**Comment by noah-paige** [comment]
Tested CDK Created Pivot Role, AWS Deployment

- [ ] CICD Pipeline Succeeds
- [ ] Update Env Stack --> Updates Pivot Role Permissions
- [ ] Create Data.all Pipeline (CDK and CodePipeline)
- [ ] Create and Delete ML Studio User Profile

File: deploy/pivot_role/pivotRole.yaml (hunk 2 above)
Comment (mourya-33) [reply deploy/pivot_role/pivotRole.yaml:217]: I merged PutLogEvents with the "Logs" SID statement set and added a prefix for lambda function log groups under resources

**Review by anmolsgandhi — APPROVED** [review]
Reviewed with @noah-paige , LGTM!

**Pull request merged**